webviz_subsurface/datainput/layeredmap/_layered_fence.py

```diff
@@ -0,0 +1,217 @@
+        return self._bounds
+
+    @property
+    def center(self):
+        '''Center of the component'''
+        return self._center
+
+    def set_bounds_and_center(self, data: [RegularSurface, Cube, tuple]):
+        '''Set bounds and center from data'''
+        if isinstance(data, RegularSurface):
+            x, y = self.slice_surface(data.copy())
+            self._bounds = [
+                [np.nanmin(x), np.nanmin(y)],
+                [np.nanmax(x), np.nanmax(y)]
+            ]
+            self._center = [np.mean(x), np.mean(y)]
+
+        elif isinstance(data, Cube):
+
+            cubefence = self.slice_cube(data)
+            self._bounds = cubefence['bounds']
+            self._center = cubefence['center']
+
+        elif (isinstance(data[0], Grid) and
+              isinstance(data[1], GridProperty)):
+            grid = data[0]
+            prop = data[1]
+            gridfence = self.slice_grid(grid, prop)
+            self._bounds = gridfence['bounds']
+            self._center = gridfence['center']
+
+        else:
+            raise TypeError(
+                'Input must be a Xtgeo surface, cube or grid/grid property')
+
+    def slice_grid(self, grid, prop, invert_y=True):
+        '''Extract line along the fencespec for the grid property'''
+
+        hmin, hmax, vmin, vmax, values = grid.get_randomline(
+            self.fencespec, prop, hincrement=self.hinc
+        )
+        if invert_y:
+            ymin = -vmax
+            ymax = -vmin
+        else:
+            ymin = vmin
+            ymax = vmax
+        bounds = [[hmin, ymin], [hmax, ymax]]
+        center = [(hmin+hmax)/2, (ymax+ymin)/2]
+
+        return {'values': values, 'bounds': bounds, 'center': center}
+
+    def slice_cube(self, cube, invert_y=True):
+        '''Extract line along the fencespec for the cube'''
+        hmin, hmax, vmin, vmax, values = cube.get_randomline(
+            self.fencespec, hincrement=self.hinc
+        )
+        if invert_y:
+            ymin = -vmax
+            ymax = -vmin
+        else:
+            ymin = vmin
+            ymax = vmax
+        bounds = [[hmin, ymin], [hmax, ymax]]
+        center = [(hmin+hmax)/2, (ymax+ymin)/2]
+
+        return {'values': values, 'bounds': bounds, 'center': center}
+
+    def slice_surface(self, surface, invert_y=True):
+        '''Extract line along the fencespec for the surface'''
+        s = surface.copy()
+        values = s.get_randomline(self.fencespec, hincrement=self.hinc)
+        x = values[:, 0]
+        y = values[:, 1]
+        if invert_y:
+            y *= -1
+        return x, y
+
+    def add_surface_layer(
+        self,
+        surface: RegularSurface,
+        name: str,
+        tooltip: str = None,
+        color: str = "blue",
+        checked: bool = True,
+    ):
+        '''Adds a polyline overlay layer
+        for a given XTGeo surface'''
+        x, y = self.slice_surface(surface.copy())
+        positions = [[a, b] for a, b in zip(x, y)]
```

Comment:
Not sure if it is more readable (or works), but could consider not introducing `a` and `b` and instead of
```python
x, y = self.slice_surface(surface.copy())
positions = [[a, b] for a, b in zip(x, y)]
```
use 
```python
positions = [[x, y] for x, y in zip(self.slice_surface(surface.copy()))]
```